phoenix-core/src/it/java/org/apache/phoenix/end2end/QueryLoggerIT.java

```diff
@@ -250,7 +250,103 @@ public void testWithLoggingOFF() throws Exception{
         assertFalse(foundQueryLog);
         conn.close();
     }
-    
+
+    @Test
+    //DROP Table statement should be always logged.
+    public void testDropTableStatement() throws Exception{
```

Comment:
Sorry for late comment but you can actually merge both the tests into on.

Reply:
Also can you test a scenario with multiple drop/alter statements issued together. What happens if the RingBuffer is full and we are trying to add to it, does it block and wait OR it drops it (in that case this is not complete)?